doc/source/serve/tutorials/gradio-integration.md

```diff
@@ -0,0 +1,128 @@
+
+We will be using [HuggingFace's Pipelines](https://huggingface.co/docs/transformers/main_classes/pipelines) to access the model. First, let's install the transformers module.
+```console
+$ pip install transformers
+```
+
+## Basic Example: Deploy your Gradio app with Ray Serve
+
+This example will show you an easy, straightforward way to deploy your app onto Ray Serve. Start by creating a new Python file named `demo.py` and import `GradioServer` from Ray Serve for deploying your Gradio app, `gradio`, and `transformers.pipeline` for loading text summarization models.
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_import_begin__
+:end-before: __doc_import_end__
+```
+
+Then, we construct the (optional) Gradio app `io`:
+:::{note} 
+Remember you can substitute this with your own Gradio app if you want to try scaling up your own Gradio app!
+:::
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_gradio_app_begin__
+:end-before: __doc_gradio_app_end__
+```
+
+
+### Understanding `GradioServer`
+In order to deploy your Gradio app onto Ray Serve, you need to wrap your Gradio app in a Serve [deployment](serve-key-concepts-deployment). `GradioServer` acts as that wrapper. It serves your Gradio app remotely on Ray Serve so that it can process and respond to HTTP requests.
+:::{note} 
+`GradioServer` is simply `GradioIngress` but wrapped in a Serve deployment.
+:::
+```{literalinclude} ../../../../python/ray/serve/gradio.py
+:start-after: __doc_gradio_ingress_begin__
+:end-before: __doc_gradio_ingress_end__
+```
+
+### Deploy your Gradio Server
+Replicas in a deployment are copies of your program living on Ray Serve, and more replicas means your deployment can serve more client requests. You can increase the number of replicas of your application or increase the number of CPUs and/or GPUs available to each replica.
+
+Then, using either the example we created above, or an existing Gradio app (of type `Interface`, `Block`, `Parallel`, etc.), wrap it in your Gradio Server.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_app_begin__
+:end-before: __doc_app_end__
+```
+
+Finally, deploy your Gradio Server! Run the following in your terminal:
+```console
+$ serve run demo:app
+```
+
+Now you can access your Gradio app at `http://localhost:8000`! This is what it should look like:
+![Gradio Result](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/gradio_result.png)
+
+See [Putting Ray Serve Deployment Graphs in Production](https://docs.ray.io/en/master/serve/production.html#id1) for more information on how to deploy your app in production.
+
+
+## Complex Example: Parallelism
```

Comment:
I wonder if there's a better way to frame these sections than "Complex Example" vs "Basic Example."

How about instead of "Basic Example: Deploy your Gradio app with Ray Serve", we say "Quickstart: Deploy your Gradio app with Ray Serve"? And instead of "Complex Example: Parallelism", we say "Parallelizing models with Ray Serve"?

Reply:
Sounds good to me! Thanks for the suggestion!